functions.lua

```diff
@@ -396,8 +402,29 @@ function tp.tpr_deny(name)
 			chat2.send_message(minetest.get_player_by_name(name2), S("Teleport request denied."), 0xFFFFFF)
 			chat2.send_message(minetest.get_player_by_name(name), S("You denied the request @1 sent you.", name2), 0xFFFFFF)
 		end
+
 		tp.tphr_list[name] = nil
 
+		-- Don't allow re-denying requests.
+		tp.tpn_list[name2] = nil
+
+	elseif tp.tpn_list[name] then
+		name2 = tp.tpn_list[name]
+		minetest.chat_send_player(name, S("You denied your request sent to @1.", name2))
+		minetest.chat_send_player(name2, S("@1 denied his request sent to you.", name))
```

Comment:
You should probably use a gender-neutral pronoun.

```suggestion
		minetest.chat_send_player(name2, S("@1 denied their request sent to you.", name))
```